cdc/sink/codec/canal_flat.go

```diff
@@ -504,5 +503,14 @@ func canalFlatMessage2DDLEvent(flatDDL canalFlatMessageInterface) *model.DDLEven
 	// we lost DDL type from canal flat json format, only got the DDL SQL.
 	result.Query = flatDDL.getQuery()
 
+	// set the DDL Type to be compatible with mysql sink's logic
+	// see https://github.com/pingcap/tiflow/blob/0578db337d783643cfab9f25ccb28a5dd0de5806/cdc/sink/mysql.go#L362-L370
+	// see https://github.com/pingcap/tidb/blob/6dbf2de2f05bbdeff6bde45d8cbad740a361dd6c/parser/model/ddl.go#L101-L102
+	if strings.HasPrefix(result.Query, "create schema") {
+		result.Type = timodel.ActionCreateSchema
```

Comment:
What was its original type?

Reply:
no ddl type information was encoded by the encoder. 

we do this to hack the decoded `RowChangedEvent`, then can be consumed by `MySQLSink`.

Reply:
no ddl type information was encoded by the encoder. 

we do this to hack the decoded `RowChangedEvent`, then can be consumed by `MySQLSink`.

Reply:
Is it case-sensitive?